src/app/GitUI/CommandsDialogs/SettingsDialog/Plugins/PluginSettingsPage.cs

```diff
@@ -17,11 +17,18 @@ public PluginSettingsPage(IServiceProvider serviceProvider)
 
         private void CreateSettingsControls()
         {
-            IEnumerable<ISetting> settings = GetSettings();
+            try
+            {
+                IEnumerable<ISetting> settings = GetSettings();
 
-            foreach (ISetting setting in settings)
+                foreach (ISetting setting in settings)
+                {
+                    AddSettingControl(setting.CreateControlBinding());
+                }
+            }
+            catch (Exception ex)
             {
-                AddSettingControl(setting.CreateControlBinding());
+                throw new Exception($"Cannot load settings for plugin {_gitPlugin?.Name ?? "unknown"}", ex);
```

Comment:
❔
```suggestion
                throw new ExternalOperationException($"Cannot load settings for plugin {_gitPlugin?.Name ?? "unknown"}", ex);
```

Reply:
This is not really an external exception, all info is not meaningful:
```
public ExternalOperationException(
    string? command = null,
    string? arguments = null,
    string? workingDirectory = null,
    int? exitCode = null,
    Exception? innerException = null)
```

UserExternalOperationException can be used, but does not make sense either to me.

Reply:
The key is that the `BugReportReportInvoker` presents the `ExternalOperationException` without "Report bug" as default but with the hint that something external failed - most likely in user's responsibility.
This might be applicable here: A plugin might be outdated.
The arguments are optional. My suggestion was incomplete.
```suggestion
                throw new ExternalOperationException(command: $"Load settings for plugin {_gitPlugin?.Name ?? "unknown"}", innerException: ex);
```

Reply:
ok
This problem is likely in the bundled plugins or GE core, but no need to alarm users...
(Also keeping Cannot)